walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
 
-public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+public final class FindMeetingQuery {        
+
+    public Collection <TimeRange> query(Collection <Event> events, MeetingRequest request) {
+      long calculation = 0;
+      long duration = request.getDuration();
+      Collection<String> mandatoryAttendees = request.getAttendees();
+      ArrayList<TimeRange> availableTimeForAll = new ArrayList<TimeRange>();
+      int counter = 0;
+      int end = TimeRange.END_OF_DAY;
+      TimeRange meetingDuration;
+      int meetingStart = 0;
+      int meetingEnd = 0;
+      int len = events.size();
+      int start = TimeRange.START_OF_DAY;
+
+      
+      if(mandatoryAttendees.isEmpty()){
+        availableTimeForAll.add(TimeRange.WHOLE_DAY);
+        return availableTimeForAll;
+      }
+      if (duration > 60 * 24){
+          return availableTimeForAll;
+      }
+
+      Set<String> attendeesSet = new HashSet<String>();
+      for(String attendee : mandatoryAttendees) {
+            attendeesSet.add(attendee);
+      }
+      int setSize = attendeesSet.size();
+
+      if(len == 0) {
+        availableTimeForAll.add(TimeRange.WHOLE_DAY);
+        return availableTimeForAll;
+      }
+
+      for(Event event: events) {
+        Collection<String> eventAttendees = event.getAttendees();
+        TimeRange when = event.getWhen();        
+
+        //if on first event
+        if(counter == 0) {
+            meetingStart = when.start();
+            String attendee1 = attendeesSet.iterator().next();
+            String eventAttendeePerson =  eventAttendees.iterator().next();
+
+            //if attendee for event is not equal to meeting request attendee then make the availableTime all day
+            if(eventAttendeePerson != attendee1){
+                availableTimeForAll.add(TimeRange.WHOLE_DAY);
```

Comment:
I'm not clear on what your intent here is (and suspect there's some mistake here);

The code seems to get the name of the first event member and in the mandatory attendees, and then compares them;
Even if both sets are identical, however, there's no guarantee that iterator().next() will return the same ones in the correct order.. Could you clarify what you meant here?